docs/vtex-io/Perks/vtex-io-documentation-homologation-requirements-for-vtex-app-store/vtex-io-documentation-engineering-guidelines.md

```diff
@@ -8,108 +8,138 @@ updatedAt: "2022-12-13T20:17:44.370Z"
 
-We value clear, well-formatted and easy-to-read codes. Therefore, please follow the criteria listed below before submitting your app:
+### Using app token for a user-initiated action
 
-- Remove unnecessary comments from the code.
-- Remove any commands used for debugging, such as `console.log()` — it should not be used in production workspaces.
-- Always use the `import` command instead of `require` to import references into Typescript files.
-- Always use the `const` command instead of `let` or `var` to declare elements in Typescript files.
-- Follow the order below when importing references into Typescripts files:
-    1. External dependencies;
-    2. Other VTEX apps;
-    3. Local modules of your code.
+For user-initiated actions, apps should use either `ctx.vtex.storeUserAuthToken` or `ctx.vtex.adminUserAuthToken` (as opposed to `ctx.authToken`). For more information, see [Connecting to VTEX Core Commerce APIs](https://developers.vtex.com/vtex-developer-docs/docs/how-to-connect-with-vtex-core-commerce-apis-using-vtex-io#steps).
 
-- Avoid implementing routes and resolvers directly in the `index.ts` file. Instead, implement the middleware functions in other files and import them into the main file
+### Declaring broad policies
 
-> ℹ️ Check out the free [VTEX IO](https://learn.vtex.com/) courses. In these course, you will find technical requirements of the development platform, examples, do's and don'ts, and answers to the main questions that may arise along the development journey. They will guide your learning at the code level, complementing the guidelines outlined in this section.
+Outbound access policies (`outbound-access`) to VTEX resources should follow the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). In short, you only give access to what is really needed. For instance: the `path` should not be configured as `*`. See an example [here](https://github.com/vtex-apps/store-graphql/blob/684dcbbbd6e9cdbd121afd7802200856cb952d2b/manifest.json#L107-L112) of how it should be done.
 
-### VTEX builders
+### Avoid exposing private information through public routes
 
-[Builders](https://developers.vtex.com/docs/guides/vtex-io-documentation-builders) work as an API responsible for configuring other IO services in your app.
-For developing and publishing apps for the VTEX App Store, the following builders must be up-to-date in your app's code:
+Public routes do not require authentication when called from the Frontend, so they should not expose information obtained from calling private APIs, e.g.: exposing a list of orders to the Frontend. A common mistake is leaving routes which are meant to only be consumed from the Admin unrestricted. Also, GraphQL APIs are public by default, so the schema should not have queries exposing private information either.
```

Comment:
```suggestion
As the name suggests, public routes do not require authentication when accessed from the frontend. Consequently, they should never reveal data sourced from private APIs, such as displaying a list of orders directly to the frontend. In this sense, a common mistake is failing to restrict routes intended exclusively for Admin usage. Additionally, keep in mind that GraphQL APIs are public by default, necessitating cautious schema design to avoid queries that inadvertently expose confidential information.
```